pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -876,6 +861,15 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 			// On failure, return error. Caller will log and retry.
 			return volumeToMount.GenerateError("MapVolume.MapDevice failed", mapErr)
 		}
+
+		// Update actual state of world to reflect volume is globally mounted
```

Comment:
@jingxu97 
Instead of moving actualStateOfWorld.MarkDeviceAsMounted() out from `if volumeAttacher != nil {} clause`, can we call `GetGlobalMapPath()` in the above if clause?

In previous, only attachable plugins are marked DeviceAsMounted, but after this change, all plugins including non-attachable plugins will be marked DeviceAsMounted too. I guess GetVolumesInUse() only handles attachable plugin's volume, so this change doesn't have impact for this, but this fix changes operation flow between Filesystem volume and Block volume.

if globalMapPath is empty before calling `og.blkUtil.MapDevice()`, we need to call GetGlobalMapPath() for non-attachable volume plugin.

Any thought?

Reply:
I can do that. Could you please let me know which plugin has attacher, which one not for this MapVolume function?

Reply:
got it. I will update the code in a min.